com.sap.cloud.lm.sl.cf.client/src/main/java/com/sap/cloud/lm/sl/cf/client/util/StreamManager.java

```diff
@@ -0,0 +1,93 @@
+package com.sap.cloud.lm.sl.cf.client.util;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.text.MessageFormat;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+
+import org.apache.commons.io.IOUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class StreamManager {
+    
+    private static final Logger LOGGER = LoggerFactory.getLogger(StreamManager.class);
+    private InputStream inputStream;
+    public static final String ARCHIVE_ENTRY_SEPARATOR = "/";
+
+    public StreamManager(InputStream inputStream) {
+        this.inputStream = inputStream;
+    }
+
+    public File saveStreamToFile(String entryName) throws IOException {
+        int i = entryName.lastIndexOf('.');
+        String fileName = (i > 0) ? entryName.substring(0, i) : entryName;
+        String suffix = (i > 0) ? entryName.substring(i) : null;
+        if (fileName.length() < 3) {
+            fileName = fileName + "000";
+        }
+        File file = File.createTempFile(fileName, suffix);
+        LOGGER.debug(MessageFormat.format("Saving input stream to temporary file \"{0}\"...", file.getPath()));
+        try (FileOutputStream outputStream = new FileOutputStream(file)) {
+            IOUtils.copy(inputStream, outputStream);
+        }
+        LOGGER.debug(MessageFormat.format("Input stream saved to temporary file \"{0}\"", file.getPath()));
+        return file;
+    }
+
+    public void saveStreamToDirectory(String entryName, String directoryName, Path tempDirectory) throws IOException {
+        StreamUtil.validateEntry(entryName);
+        Path filePath = StreamUtil.resolveTempEntryPath(entryName, directoryName, tempDirectory);
+        StreamUtil.createParentDirectories(filePath);
+        createFile(filePath);
+    }
+
+    public File saveZipStreamToDirectory(String entryName, long maxZipEntrySize) throws IOException {
+        Path path = StreamUtil.getTempDirectory(entryName);
+        File dir = path.toFile();
+        LOGGER.debug(MessageFormat.format("Saving input stream to temporary directory \"{0}\"...", dir.getPath()));
+        saveEntries(path, entryName, maxZipEntrySize);
+        LOGGER.debug(MessageFormat.format("Input stream saved to temporary directory \"{0}\"", dir.getPath()));
+        return dir;
+    }
+
+    private void saveEntries(Path tempDirectory, String rootEntryName, long maxZipEntrySize) throws IOException, FileNotFoundException {
+        ZipInputStream zis = (ZipInputStream) inputStream;

```

Comment:
zis is too algorithmic. Could we rename it it archiveInputStream or smthing similar?